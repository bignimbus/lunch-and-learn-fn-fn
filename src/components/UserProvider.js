import React, { Component } from 'react';
import PropTypes from 'prop-types';
import UserService from '../services/UserService';

class UserProvider extends Component {
  static propTypes = {
    WrappedComponent: PropTypes.func.isRequired,
  };

  state = {
    user: {},
  };

  componentDidMount () {
    this.getUser();
  }

  async getUser () {
    const user = await UserService.getUser();
    this.setState({ user });
  }

  render () {
    const { user } = this.state;
    const { WrappedComponent } = this.props;

    return (
      <div>
        {
          user ?
            <WrappedComponent user={user} /> :
            <span>Loading...</span>
        }
      </div>
    );
  }
}

export default UserProvider;
