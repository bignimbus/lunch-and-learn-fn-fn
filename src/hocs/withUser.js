import React from 'react';
import UserProvider from '../components/UserProvider';

const withUser = (WrappedComponent) => () => (
  <UserProvider WrappedComponent={WrappedComponent} />
);

export default withUser;
