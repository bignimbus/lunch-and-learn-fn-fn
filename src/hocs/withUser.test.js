import React from 'react';
import renderer from 'react-test-renderer';
import withUser from './withUser';
import UserProvider from '../components/UserProvider';

const TestComponent = () => <header id='my-hoc-test' />;

describe('withUser', () => {
  it('should return a composed component', () => {
    const TestComponentContainer = withUser(TestComponent);
    const component = renderer.create(<TestComponentContainer />);
    expect(component.root.findByType(UserProvider).instance).toBeInstanceOf(UserProvider);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
