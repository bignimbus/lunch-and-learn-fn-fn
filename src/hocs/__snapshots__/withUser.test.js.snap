// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withUser should return a composed component 1`] = `
<div>
  <header
    id="my-hoc-test"
  />
</div>
`;
